app/src/main/java/com/example/myapplication/ThirdFragment.java

```diff
@@ -36,35 +43,150 @@ public class ThirdFragment extends Fragment {
+        v = container;
+        ColorStateList purple = ColorStateList.valueOf(Color.parseColor(("#6750A3")));
+        ColorStateList blue = ColorStateList.valueOf((Color.parseColor("#73C2F0")));
+        if (binding.switch1.isChecked()){
+            binding.switch1.setThumbTintList(purple);
+            binding.switch1.setTrackTintList(purple);
+        } else {
+            binding.switch1.setThumbTintList(blue);
+            binding.switch1.setTrackTintList(blue);
+        }
+        binding.switch2.setChecked(MainActivity.defense);
+        if (binding.switch2.isChecked()){
+            binding.switch2.setThumbTintList(blue);
+            binding.switch2.setTrackTintList(blue);
+        } else {
+            binding.switch2.setThumbTintList(blue);
+            binding.switch2.setTrackTintList(blue);
+        }
+        binding.switch3.setChecked(MainActivity.ground);
+        if (binding.switch3.isChecked()){
+            binding.switch3.setThumbTintList(blue);
+            binding.switch3.setTrackTintList(blue);
+        } else {
+            binding.switch3.setThumbTintList(blue);
+            binding.switch3.setTrackTintList(blue);
+        }
+
         return binding.getRoot();
     }
 
     public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
+        ColorStateList purple = ColorStateList.valueOf(Color.parseColor(("#6750A3")));
+        ColorStateList blue = ColorStateList.valueOf((Color.parseColor("#73C2F0")));
         binding.toAutoCont.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
                 NavHostFragment.findNavController(ThirdFragment.this)
                         .navigate(R.id.action_ThirdFragment_to_thirdFragment2);
             }
         });
+        ObjectAnimator animation = ObjectAnimator.ofFloat(binding.pop, "rotationX", new float[]{0f, 90f, 180f, 270f, 360f, 90f, 180f, 270f, 360f, 90f, 180f, 270f, 360f});
+        animation.setDuration(1000);
+        binding.pop.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                animation.start();
+                MainActivity.darkMode = !MainActivity.darkMode;
+                lightDark(v, MainActivity.darkMode);
+            }
+        });
+        binding.switch1.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked)  {
+                if (binding.switch1.isChecked()){
+                    binding.switch1.setThumbTintList(purple);
+                    binding.switch1.setTrackTintList(purple);
+                } else {
+                    binding.switch1.setThumbTintList(blue);
+                    binding.switch1.setTrackTintList(blue);
+                }
```

Comment:
This is repeated multiple times; consider creating a new method, similar to the one below:

    private void setSwitchColor(Switch switch) {
        if (switch.isChecked()) {
            switch.setThumbTintList(purple);
            switch.setTrackTintList(purple);
        } else {
            switch.setThumbTintList(blue);
            switch.setTrackTintList(blue);
        }
    }

And calling here as `this.setSwitchColor(binding.switch1)` (and similar in other places)

Reply:
fixed

Reply:
fixed